pylot/perception/detection/detection_operator.py

```diff
@@ -40,42 +40,20 @@ def __init__(self, camera_stream: erdos.ReadStream,
         self._logger = erdos.utils.setup_logging(self.config.name,
                                                  self.config.log_file_name)
         self._obstacles_stream = obstacles_stream
-        self._detection_graph = tf.Graph()
-        # Load the model from the model file.
-        pylot.utils.set_tf_loglevel(logging.ERROR)
-        with self._detection_graph.as_default():
-            od_graph_def = tf.GraphDef()
-            with tf.gfile.GFile(model_path, 'rb') as fid:
-                serialized_graph = fid.read()
-                od_graph_def.ParseFromString(serialized_graph)
-                tf.import_graph_def(od_graph_def, name='')
-
-        self._gpu_options = tf.GPUOptions(
-            allow_growth=True,
-            visible_device_list=str(self._flags.obstacle_detection_gpu_index),
-            per_process_gpu_memory_fraction=flags.
-            obstacle_detection_gpu_memory_fraction)
-        # Create a TensorFlow session.
-        self._tf_session = tf.Session(
-            graph=self._detection_graph,
-            config=tf.ConfigProto(gpu_options=self._gpu_options))
-        # Get the tensors we're interested in.
-        self._image_tensor = self._detection_graph.get_tensor_by_name(
-            'image_tensor:0')
-        self._detection_boxes = self._detection_graph.get_tensor_by_name(
-            'detection_boxes:0')
-        self._detection_scores = self._detection_graph.get_tensor_by_name(
-            'detection_scores:0')
-        self._detection_classes = self._detection_graph.get_tensor_by_name(
-            'detection_classes:0')
-        self._num_detections = self._detection_graph.get_tensor_by_name(
-            'num_detections:0')
+
+        physical_devices = tf.config.experimental.list_physical_devices('GPU') 
+        # Equivalent to allow_growth=True
+        for device in physical_devices:
+            tf.config.experimental.set_memory_growth(device, True)
```

Comment:
Can we please check the device index flag before setting the memory growth? Our earlier configurations allowed developers to deploy models on specific GPUs using `FLAGS.obstacle_detection_gpu_index`, which I think is missing in the new operator?

Reply:
Also, do you see huge logs from tensorflow? If yes, we should probably check if `pylot.utils.set_tf_loglevel` is still valid and invoke it. If not, we should deprecate/remove that util.

Reply:
`pylot.utils.set_tf_loglevel(logging.ERROR)` does not seem to work anymore for limiting the TF logs

Reply:
Adjusted this to only set memory growth for the GPU that is flagged by `FLAGS.obstacle_detection_gpu_index`